app/controllers/books_controller.rb

```diff
@@ -0,0 +1,55 @@
+# Controller to handle book management
+class BooksController < ApplicationController
+  def show
+    book = Book.find_by(id: params[:id])
+    if book
+      render json: book, status: :ok
+    else
+      render json: nil, status: :not_found
+    end
+  end
+
+  def index
+    books = Book.search(params).page(params[:page]).per(params[:per_page])
+    render json: books, status: :ok
+  end
+
+  def create
+    book = Book.new(book_params)
+    authorize book
+    if book.save
+      render json: book, status: :created
+    else
+      render json: { errors: book.errors }, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    book = Book.find(params[:id])
+    authorize book
+    if book.update(book_params)
+      render json: book, status: :ok
+    else
+      render json: { errors: book.errors }, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    book = Book.find_by(id: params[:id])
```

Comment:
Use Book.find, don't test if book doesn't exist, if it doesn't exist it should throw  a record not found error and be catched in the application controller